### ISSUE TITLE
[Accessibility]: Closing search contexts dropdown should focus the dropdown trigger

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43129]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188

### Problem description

When closing the search context dropdown menu with the Escape key, the close ("X") button, or the "Reset" button, nothing is focused or the search box is focused. 

### Expected behavior

The search context dropdown button should be focused after closing the menu.

![CleanShot 2022-10-18 at 14 53 24@2x](https://user-images.githubusercontent.com/206864/196552223-2417dc85-3048-42a2-8871-beb176f3291b.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

/cc @lguychard

**Comment by gitstart-sourcegraph** [comment]
Hi @limitedmage @umpox 
Would like to get some context on specific cases which/when focus should be moved to the search input relative to the reset action. At the moment we move focus to input whenever there's a change in selected search context spec which include the `reset` action.
For the reset action, should the context change behavior be unique in terms of moving focus to trigger?

**Comment by limitedmage** [comment]
@gitstart-sourcegraph Good call, thanks for pointing this out. Let's go with the following:

- Focus should be back into the dropdown when closing with no changes (eg. using `esc` key or the close button). 
- Focus should go to the search input if there are changes (eg. selecting a context from the list or using the reset button).

Does this make sense?

**Comment by gitstart-sourcegraph** [comment]
Yea that makes sense, thanks for clarifying.
